### ISSUE TITLE
Dbt ref function can take up to 2 arguments - sqlfluff supports only 1

### ISSUE THREAD

**Issue by tevariou** [sqlfluff/sqlfluff#4809]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Sqlfluff returns an error when two arguments are added to the dbt function `ref`
```jinja2
SELECT * FROM {{ ref('package_name', 'model_name') }}
```
According to the DBT documentation https://docs.getdbt.com/reference/dbt-jinja-functions/ref#two-argument-variant , it should be valid.

### Expected Behaviour

No error raised 

### Observed Behaviour

Sqlfluff raises this error:
```
Unrecoverable failure in Jinja templating: <lambda>() takes 1 positional argument but 2 were given. Have you configured your variables?
```

### How to reproduce

Lint this DBT model
```
```jinja2
SELECT * FROM {{ ref('package_name', 'model_name') }}
```

### Dialect

postgres

### Version

Latest

### Configuration

pre-commit config
```
- repo: https://github.com/sqlfluff/sqlfluff
  rev: 2.0.7
  hooks:
    - id: sqlfluff-lint
      additional_dependencies: ["dbt-postgres", "sqlfluff-templater-dbt", "pytz==2022.7.1"]
      args: [--exclude-rules, "L026, L028, L031, L034", --dialect, "postgres"]
      files: ^dbt/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
